### PR TITLE
environmentd: deflake test_cancel_long_running_query

### LINKED ISSUE #16990: `test_cancel_long_running_query` is flaky
### Buildkite link

https://buildkite.com/materialize/tests/builds/47827#018583be-7d46-44af-88fd-a46f8831df26

### Relevant log output

```source
TIMEOUT [ 240.026s]            mz-environmentd::server test_cancel_long_running_query
 
--- STDOUT:                         mz-environmentd::server test_cancel_long_running_query ---
 
running 1 test
test test_cancel_long_running_query has been running for over 60 seconds
 
--- STDERR:                         mz-environmentd::server test_cancel_long_running_query ---
2023-01-05T21:10:11.454162Z  INFO clusterd: booting os.os_type=Ubuntu os.version=22.4.0 os.bitness=64-bit build.version="0.39.0-dev" build.sha="763d0cd77e6854a9f46b1a6cebdcb5d603e5da5c" build.time="2023-01-05T21:05:51Z" cpus.logical=32 cpus.physical=16 cpu0.brand="AMD EPYC 7R32" cpu0.frequency=3291 memory.total=66706653184 memory.used=8754761728 memory.limit=9223372036854771712 swap.total=0 swap.used=0 swap.limit=0 tracing.max_level=info
2023-01-05T21:10:11.454727Z  INFO clusterd: serving internal HTTP server on /tmp/ebe74e59f33b9b431831d6aa92909411cb43de44
2023-01-05T21:10:11.468607Z  INFO clusterd: listening for storage controller connections on /tmp/a4804065865cd3c2e352a29a9276a60ab0a42754
2023-01-05T21:10:11.468915Z  INFO clusterd: listening for compute controller connections on /tmp/4b7eeec980a0cd4cf74baa98f46a829baf302f09
2023-01-05T21:10:11.468890Z  INFO mz_service::grpc: Starting to listen on /tmp/a4804065865cd3c2e352a29a9276a60ab0a42754
2023-01-05T21:10:11.469141Z  INFO mz_service::grpc: Starting to listen on /tmp/4b7eeec980a0cd4cf74baa98f46a829baf302f09
2023-01-05T21:10:11.546658Z  INFO mz_service::grpc: GrpcServer: remote client connected
2023-01-05T21:10:11.549400Z  INFO mz_compute::server: Building timely container with config CommunicationConfig { workers: 1, process: 0, addresses: ["/tmp/environmentd-local-az1-6c5cd571-ca88-4fc7-88e6-54d029225a23-0/compute-cluster-u1-replica-1/compute-0"] }
2023-01-05T21:10:11.549725Z  INFO mz_compute::communication: initializing network for multi-process timely instance, with 1 processes for epoch number (2, 1) process=0
2023-01-05T21:10:11.550274Z  INFO mz_compute::communication: successfully initialized network process=0
2023-01-05T21:10:12.252975Z  INFO clusterd: booting os.os_type=Ubuntu os.version=22.4.0 os.bitness=64-bit build.version="0.39.0-dev" build.sha="763d0cd77e6854a9f46b1a6cebdcb5d603e5da5c" build.time="2023-01-05T21:05:51Z" cpus.logical=32 cpus.physical=16 cpu0.brand="AMD EPYC 7R32" cpu0.frequency=3290 memory.total=66706653184 memory.used=8838754304 memory.limit=9223372036854771712 swap.total=0 swap.used=0 swap.limit=0 tracing.max_level=info
2023-01-05T21:10:12.253575Z  INFO clusterd: serving internal HTTP server on /tmp/7125988171def41cabec8734ac3742e5bec31a2f
2023-01-05T21:10:12.268356Z  INFO clusterd: listening for storage controller connections on /tmp/0a127b430faaaa54ef09aee50b4be089a0f8cb13
2023-01-05T21:10:12.268625Z  INFO clusterd: listening for compute controller connections on /tmp/bbd33c5c0e431048e8654871924d250f7b730fbc
2023-01-05T21:10:12.268666Z  INFO mz_service::grpc: Starting to listen on /tmp/0a127b430faaaa54ef09aee50b4be089a0f8cb13
2023-01-05T21:10:12.268859Z  INFO mz_service::grpc: Starting to listen on /tmp/bbd33c5c0e431048e8654871924d250f7b730fbc
2023-01-05T21:10:12.381892Z  INFO mz_service::grpc: GrpcServer: remote client connected
2023-01-05T21:10:12.384347Z  INFO mz_compute::server: Building timely container with config CommunicationConfig { workers: 1, process: 0, addresses: ["/tmp/environmentd-local-az1-6c5cd571-ca88-4fc7-88e6-54d029225a23-0/compute-cluster-s1-replica-2/compute-0"] }
2023-01-05T21:10:12.384643Z  INFO mz_compute::communication: initializing network for multi-process timely instance, with 1 processes for epoch number (2, 1) process=0
2023-01-05T21:10:12.385125Z  INFO mz_compute::communication: successfully initialized network process=0
2023-01-05T21:10:13.296321Z  INFO clusterd: booting os.os_type=Ubuntu os.version=22.4.0 os.bitness=64-bit build.version="0.39.0-dev" build.sha="763d0cd77e6854a9f46b1a6cebdcb5d603e5da5c" build.time="2023-01-05T21:05:51Z" cpus.logical=32 cpus.physical=16 cpu0.brand="AMD EPYC 7R32" cpu0.frequency=3292 memory.total=66706653184 memory.used=8899485696 memory.limit=9223372036854771712 swap.total=0 swap.used=0 swap.limit=0 tracing.max_level=info
2023-01-05T21:10:13.296911Z  INFO clusterd: serving internal HTTP server on /tmp/7dc1a92d59d7ea9fde2fcf2953825583c125ccae
2023-01-05T21:10:13.310816Z  INFO clusterd: listening for storage controller connections on /tmp/a4b4eba1df0809552843396f2c5cff7832de5b37
2023-01-05T21:10:13.311106Z  INFO clusterd: listening for compute controller connections on /tmp/9335b3ba26d7050f8772ab59b1011ce0adf3c52b
2023-01-05T21:10:13.311114Z  INFO mz_service::grpc: Starting to listen on /tmp/a4b4eba1df0809552843396f2c5cff7832de5b37
2023-01-05T21:10:13.311395Z  INFO mz_service::grpc: Starting to listen on /tmp/9335b3ba26d7050f8772ab59b1011ce0adf3c52b
2023-01-05T21:10:13.367395Z  INFO mz_service::grpc: GrpcServer: remote client connected
2023-01-05T21:10:13.370586Z  INFO mz_compute::server: Building timely container with config CommunicationConfig { workers: 1, process: 0, addresses: ["/tmp/environmentd-local-az1-6c5cd571-ca88-4fc7-88e6-54d029225a23-0/compute-cluster-s2-replica-3/compute-0"] }
2023-01-05T21:10:13.370949Z  INFO mz_compute::communication: initializing network for multi-process timely instance, with 1 processes for epoch number (2, 1) process=0
2023-01-05T21:10:13.371574Z  INFO mz_compute::communication: successfully initialized network process=0
2023-01-05T21:10:15.869736Z  INFO tokio_postgres::connection: NOTICE: relation "consensus" already exists, skipping
2023-01-05T21:10:15.870609Z  INFO tokio_postgres::connection: NOTICE: relation "consensus" already exists, skipping
2023-01-05T21:10:15.919183Z  INFO tokio_postgres::connection: NOTICE: relation "consensus" already exists, skipping
2023-01-05T21:10:16.036831Z  INFO tokio_postgres::connection: NOTICE: relation "consensus" already exists, skipping
2023-01-05T21:10:16.179614Z  INFO mz_persist_client::internal::machine: snapshot sfe5ab795-4654-46d4-b0fe-4114ef86c20e as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.196092Z  INFO mz_persist_client::internal::machine: snapshot s75c2380d-7d52-4197-857c-bacaba047209 as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.219049Z  INFO mz_persist_client::internal::machine: snapshot s698b2897-ebee-40eb-ae1c-11427fd9ff65 as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.241914Z  INFO mz_persist_client::internal::machine: snapshot s4d09e615-2706-408b-a015-08eb8bfaa261 as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.244466Z  INFO mz_persist_client::internal::machine: snapshot s92bb596e-32ae-40c1-a624-ef8b98b1f4cb as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.250430Z  INFO mz_persist_client::internal::machine: snapshot sfe5ab795-4654-46d4-b0fe-4114ef86c20e as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 128ms
2023-01-05T21:10:16.273190Z  INFO mz_persist_client::internal::machine: snapshot sf19c35ec-3bd2-4597-970c-892e522fdbaf as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.284048Z  INFO mz_persist_client::internal::machine: snapshot s75c2380d-7d52-4197-857c-bacaba047209 as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 128ms
2023-01-05T21:10:16.296503Z  INFO mz_persist_client::internal::machine: snapshot sa08d91ec-28e7-4a49-8721-7b3f2ccfe6aa as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.299059Z  INFO mz_persist_client::internal::machine: snapshot s698b2897-ebee-40eb-ae1c-11427fd9ff65 as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 128ms
2023-01-05T21:10:16.299248Z  INFO mz_persist_client::internal::machine: snapshot s699de774-6e1d-4b2c-83f4-725d0189f1a4 as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.300022Z  INFO mz_persist_client::internal::machine: snapshot s260fcb76-c49c-4bd7-a80f-4992c9af87ac as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.318739Z  INFO mz_persist_client::internal::machine: snapshot s461ecfa1-2214-48e0-bf03-14391328bd1f as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.324706Z  INFO mz_persist_client::internal::machine: snapshot s92bb596e-32ae-40c1-a624-ef8b98b1f4cb as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 128ms
2023-01-05T21:10:16.326411Z  INFO mz_persist_client::internal::machine: snapshot s4d09e615-2706-408b-a015-08eb8bfaa261 as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 128ms
2023-01-05T21:10:16.332864Z  INFO mz_persist_client::internal::machine: snapshot s94786fac-0faa-4bcb-b424-bc55726e2614 as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.343363Z  INFO mz_persist_client::internal::machine: snapshot sbb1ea50b-9c73-4007-b18d-6e49ac0dbdf7 as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.347005Z  INFO mz_persist_client::internal::machine: snapshot sf19c35ec-3bd2-4597-970c-892e522fdbaf as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 128ms
2023-01-05T21:10:16.360875Z  INFO mz_persist_client::internal::machine: snapshot s7a010713-a27d-4c1b-bfee-ebbdcebb0192 as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.365511Z  INFO mz_persist_client::internal::machine: snapshot s72690114-f45f-48b3-b147-9a9492367a21 as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.369275Z  INFO mz_persist_client::internal::machine: snapshot s260fcb76-c49c-4bd7-a80f-4992c9af87ac as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 128ms
2023-01-05T21:10:16.391494Z  INFO mz_persist_client::internal::machine: snapshot s461ecfa1-2214-48e0-bf03-14391328bd1f as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 128ms
2023-01-05T21:10:16.420958Z  INFO mz_persist_client::internal::machine: snapshot s6800ea8a-8978-4265-8b2a-c70cb0c4bf5c as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.489896Z  INFO mz_persist_client::internal::machine: snapshot s6800ea8a-8978-4265-8b2a-c70cb0c4bf5c as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 128ms
2023-01-05T21:10:16.513877Z  INFO mz_persist_client::internal::machine: snapshot s1db82aa1-1226-4efd-95c9-1a2c5cde3aab as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.514045Z  INFO mz_persist_client::internal::machine: snapshot se07012b9-a3e6-490e-9d4a-8522cba7704d as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.641793Z  INFO mz_persist_client::internal::machine: snapshot s1db82aa1-1226-4efd-95c9-1a2c5cde3aab as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 128ms
2023-01-05T21:10:16.641955Z  INFO mz_persist_client::internal::machine: snapshot se07012b9-a3e6-490e-9d4a-8522cba7704d as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 128ms
2023-01-05T21:10:16.642078Z  INFO mz_persist_client::internal::machine: snapshot s5c9277e1-af38-4d3c-a4ff-19e0201d28cd as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.697464Z  INFO mz_persist_client::internal::machine: snapshot s6800ea8a-8978-4265-8b2a-c70cb0c4bf5c as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 256ms
2023-01-05T21:10:16.701945Z  INFO mz_persist_client::internal::machine: snapshot s03f3aff9-8292-4e70-8d09-e0426d129092 as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.721477Z  INFO mz_persist_client::internal::machine: snapshot s5c9277e1-af38-4d3c-a4ff-19e0201d28cd as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 128ms
2023-01-05T21:10:16.743402Z  INFO mz_persist_client::internal::machine: snapshot s53d7501f-eef7-486a-ae4f-f4f3a7fe33f3 as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.783700Z  INFO mz_persist_client::internal::machine: snapshot s1db82aa1-1226-4efd-95c9-1a2c5cde3aab as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 256ms
2023-01-05T21:10:16.784913Z  INFO mz_persist_client::internal::machine: snapshot s03f3aff9-8292-4e70-8d09-e0426d129092 as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 128ms
2023-01-05T21:10:16.811533Z  INFO mz_persist_client::internal::machine: snapshot se7f0ef4d-8630-4c62-8591-fcd3a5393b69 as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.826396Z  INFO mz_persist_client::internal::machine: snapshot se07012b9-a3e6-490e-9d4a-8522cba7704d as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 256ms
2023-01-05T21:10:16.841481Z  INFO mz_persist_client::internal::machine: snapshot s53d7501f-eef7-486a-ae4f-f4f3a7fe33f3 as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 128ms
2023-01-05T21:10:16.841769Z  INFO mz_persist_client::internal::machine: snapshot s4933f28b-2f3d-48fe-a2a0-60cb213103f2 as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 64ms
2023-01-05T21:10:16.904943Z  INFO mz_persist_client::internal::machine: snapshot s5c9277e1-af38-4d3c-a4ff-19e0201d28cd as of Antichain { elements: [0] } not yet available for upper Antichain { elements: [0] } retrying in 256ms
2023-01-05T21:10:20.902769Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953020179] } retrying in 64ms
2023-01-05T21:10:20.971967Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953020179] } retrying in 128ms
2023-01-05T21:10:21.109757Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953020747] } retrying in 256ms
2023-01-05T21:10:21.373968Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953020937] } retrying in 512ms
2023-01-05T21:10:21.850388Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953020937] } retrying in 1.024s
2023-01-05T21:10:22.854822Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953022108] } retrying in 2.048s
2023-01-05T21:10:24.746975Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953024107] } retrying in 4.096s
2023-01-05T21:10:29.059078Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953028937] } retrying in 8.192s
2023-01-05T21:10:38.007771Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953037108] } retrying in 16s
2023-01-05T21:10:52.801192Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953052396] } retrying in 16s
2023-01-05T21:11:07.756298Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953067107] } retrying in 16s
2023-01-05T21:11:22.699366Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953082397] } retrying in 16s
2023-01-05T21:11:37.552938Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953097107] } retrying in 16s
2023-01-05T21:11:54.743740Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953114108] } retrying in 16s
2023-01-05T21:12:11.694070Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953131107] } retrying in 16s
2023-01-05T21:12:28.578589Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953148107] } retrying in 16s
2023-01-05T21:12:45.075479Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953164108] } retrying in 16s
2023-01-05T21:13:00.544175Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953180107] } retrying in 16s
2023-01-05T21:13:17.923549Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953197108] } retrying in 16s
2023-01-05T21:13:33.398226Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953212937] } retrying in 16s
2023-01-05T21:13:49.917786Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953229108] } retrying in 16s
2023-01-05T21:14:04.927857Z  INFO mz_persist_client::internal::machine: snapshot sd57aaa4e-fcb3-411c-b50f-9351f7852ef4 as of Antichain { elements: [18446744073709551615] } not yet available for upper Antichain { elements: [1672953242400] } retrying in 16s
```

### Additional thoughts

Test started failing after https://github.com/MaterializeInc/materialize/pull/16949 merged.

### PR DESCRIPTION
**Pull request by mjibson**
The delayed startup increased the existing flakiness of this test because there is now less time between ack'ing the connect and attempting to cancel. Fix twice by waiting for the CREATE to finish, and then cancelling in a loop.

Fixes #16990

### Motivation

  * This PR fixes a recognized bug.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - n/a

### CODE REVIEW

File: src/environmentd/tests/server.rs [hunk 1]
```diff
@@ -359,14 +359,17 @@ fn test_cancel_long_running_query() {
...
     thread::spawn(move || {
-        // Abort the query after 2s.
-        thread::sleep(Duration::from_secs(2));
-        let _ = cancel_token.cancel_query(postgres::NoTls);
+        // Repeatedly attempt to abort the query because we're not sure exactly
+        // when the SELECT will arrive.
+        loop {
+            thread::sleep(Duration::from_secs(1));
+            let _ = cancel_token.cancel_query(postgres::NoTls);
+        }
```
Comment (benesch) [inline src/environmentd/tests/server.rs:370]: I think this thread will leak after the test finishes, which seems not great. Also slight chance that it cancels the query on 379 incorrectly. I think we need a little synchronization to make sure the thread shuts down on L378. Probably easiest to switch it to a Tokio task, abort it on L378, and then await the handle, to make sure it's good and truly gone?

**Review by benesch — COMMENTED** [review]
Ooh, nice catch!

File: src/environmentd/tests/server.rs (hunk 1 above)
Comment (mjibson) [reply src/environmentd/tests/server.rs]: Ah yes. Used not tokio instead because it seemed easier than dealing with a runtime.

**Review by benesch — APPROVED** [review]
Ah yeah, this seems great too, ty!

**Pull request merged**